File: java/src/main/java/mx/iteso/adapter/word/WordAdapter.java
package mx.iteso.adapter.word;

import mx.iteso.adapter.BackgroundImage;
import mx.iteso.adapter.Format;
import mx.iteso.adapter.googledocs.IGoogleDoc;

import java.awt.Font;

/**
 * Word Adapter.
 */
public class WordAdapter  implements IGoogleDoc {

    /**
     * Word Doc.
     */
    private IWordDocument wordDocument;

    /**
     *
     * @param newWordDoc .
     */
    public WordAdapter(final WordDocument newWordDoc) {
        this.wordDocument = newWordDoc;
    }

    /**
     *
     * @return font.
     */
    public Font getFont() {
        return this.wordDocument.getFormat().getFont();
    }

    /**
     *
     * @return style format.
     */
    public Format getStyle() {
        return this.wordDocument.getFormat();
    }

    /**
     *
     * @return image backgropuind.
     */
    public BackgroundImage getBackground() {
        return new BackgroundImage(this.wordDocument.getBackground());
    }

    /**
     *
     * @param sharingPermissions .
     */
    public void setSharingPermissions(final int sharingPermissions) {
        return;
    }

    /**
     *
     * @return if license is valid.
     */
    public int getSharingPermissions() {
        return this.wordDocument.restrictEditIfLicenseIsInvalid() ? 0 : 1;
    }
}

File: csharp/AdapterPractice/AdapterPractice/Adapter/Word/WordAdapter.cs
using System;
using System.Collections.Generic;
using System.Text;
using AdapterPractice.Adapter.GoogleDocs;

namespace AdapterPractice.Adapter.Word
{
    public class WordAdapter : IGoogleDoc{

   
    private IWordDocument wordDocument;


    public WordAdapter(WordDocument newWordDoc)
    {
        this.wordDocument = newWordDoc;
    }


    public Font getFont()
    {
        return this.wordDocument.getFormat().getFont();
    }


    public Format getStyle()
    {
        return this.wordDocument.getFormat();
    }

    public BackgroundImage getBackground()
    {
        return new BackgroundImage(this.wordDocument.getBackground());
    }

    public void setSharingPermissions(int sharingPermissions)
    {
        return;
    }


    public int getSharingPermissions()
    {
        return this.wordDocument.restrictEditIfLicenseIsInvalid() ? 0 : 1;
    }
    }

}
